py4j-java/src/main/java/py4j/StringUtil.java

```diff
@@ -33,16 +33,20 @@
  * <p>
  * String utility class providing operations to escape and unescape new lines.
  * </p>
- * 
+ *
  * @author Barthelemy Dagenais
- * 
+ *
  */
 public class StringUtil {
 
 	public final static char ESCAPE_CHAR = '\\';
 
 	public static String escape(String original) {
-		return original.replace("\\", "\\\\").replace("\r", "\\r").replace("\n", "\\n");
+		if (original != null) {
```

Comment:
why is it necessary to add a check here considering that unescape has no such check?

Reply:
I only use `escape` for the auth token, since it's a string and someone might add a new line there and break things. I don't need unescape for it.

This avoids adding another method to do a null-safe escape.